RecoEgamma/PhotonIdentification/plugins/PhotonMVAEstimatorRunIIFall17.cc

```diff
@@ -0,0 +1,312 @@
+#include "RecoEgamma/PhotonIdentification/plugins/PhotonMVAEstimatorRunIIFall17.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "TMVA/MethodBDT.h"
+#include <vector>
+
+PhotonMVAEstimatorRunIIFall17::PhotonMVAEstimatorRunIIFall17(const edm::ParameterSet& conf):
+  AnyMVAEstimatorRun2Base(conf),
+  MethodName_("BDTG method"),
+  phoChargedIsolationLabel_(conf.getParameter<edm::InputTag>("phoChargedIsolation")), 
+  phoPhotonIsolationLabel_(conf.getParameter<edm::InputTag>("phoPhotonIsolation")), 
+  phoWorstChargedIsolationLabel_(conf.getParameter<edm::InputTag>("phoWorstChargedIsolation")), 
+  rhoLabel_(conf.getParameter<edm::InputTag>("rho")),
+  effectiveAreas_( (conf.getParameter<edm::FileInPath>("effAreasConfigFile")).fullPath()),
+  phoIsoPtScalingCoeff_(conf.getParameter<std::vector<double >>("phoIsoPtScalingCoeff"))
+  // phoIsoCutoff_(conf.getParameter<double>("phoIsoCutoff"))
+{
+  
+  //
+  // Construct the MVA estimators
+  //
+  tag_ = conf.getParameter<std::string>("mvaTag");
+  
+  const std::vector <std::string> weightFileNames
+    = conf.getParameter<std::vector<std::string> >("weightFileNames");
+  
+  if( weightFileNames.size() != nCategories )
+    throw cms::Exception("MVA config failure: ")
+      << "wrong number of weightfiles" << std::endl;
+  
+  gbrForests_.clear();
+  // The method name is just a key to retrieve this method later, it is not
+  // a control parameter for a reader (the full definition of the MVA type and
+  // everything else comes from the xml weight files).
+  
+  // Create a TMVA reader object for each category
+  for(uint i=0; i<nCategories; i++){
+    
+    // Use unique_ptr so that all readers are properly cleaned up
+    // when the vector clear() is called in the destructor
+    
+    edm::FileInPath weightFile( weightFileNames[i] );
+    gbrForests_.push_back( createSingleReader(i, weightFile ) );
+    
+  }
+  
+}
+
+PhotonMVAEstimatorRunIIFall17::
+~PhotonMVAEstimatorRunIIFall17(){
+}
+
+float PhotonMVAEstimatorRunIIFall17::
+mvaValue(const edm::Ptr<reco::Candidate>& particle, const edm::Event& iEvent) const {  
+  
+  const int iCategory = findCategory( particle );
+
+  const std::vector<float> vars = fillMVAVariables( particle, iEvent ) ; 
+  const float result = gbrForests_.at(iCategory)->GetClassifier(vars.data());
+  
+  // DEBUG
+  const bool debug = false;
```

Comment:
why not make this configurable so that the code does not need to be recompiled?

Reply:
Need to ask to the analyzer who implemented this ID to do that.